### PR TITLE
Workspace: casr & libcasr

### LINKED ISSUE #39: Create workspace for libcasr and casr (binaries)

### PR DESCRIPTION
**Pull request by anfedotoff**
Closes #39

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,40 +1,11 @@
...
-linux-personality = "~1.0"
-thiserror = "~1.0"
-colored = "~2.0"
+[workspace]
+members = ["casr", "libcasr"]
 
 [profile.release]
+lto = true
 opt-level = 3
+debug = true
```
Comment (SweetVishnya) [inline Cargo.toml:11]: disable debug in release

File: casr/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+rayon = "~1.5"
+num_cpus = "~1.13"
+linux-personality = "~1.0"
+colored = "~2.0"
+serde = { version = "~1.0", features = ["derive"] }
+serde_json = "~1.0"
+regex = "~1.5"
+libc = "~0"
+
+libcasr = { path = "../libcasr", version = "2.4.0", features = ["serde", "exploitable"] }
```
Comment (SweetVishnya) [inline casr/Cargo.toml:33]: do we need to specify version here?

Comment (SweetVishnya) [reply casr/Cargo.toml:33]: It will complicate making releases

File: casr/src/bin/casr-san.rs [hunk 3]
```diff
@@ -187,16 +187,20 @@ fn main() -> Result<()> {
             if let Some(signal) = sanitizers_result.status.signal() {
                 // Get stack trace and mappings from gdb.
                 match signal as u32 {
-                    SIGINFO_SIGILL | SIGINFO_SIGSYS => {
+                    // SIGILL, SIGSYS
+                    4 | 31 => {
```
Comment (SweetVishnya) [inline casr/src/bin/casr-san.rs:191]: Why did we remove the constants?

File: libcasr/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,27 @@
+[package]
+name = "libcasr"
+authors = ["Andrey Fedotov  <fedotoff@ispras.ru>", "Alexey Vishnyakov <vishnya@ispras.ru>", "Georgy Savidov <avgor46@ispras.ru>"]
+version = "2.4.0"
```
Comment (SweetVishnya) [inline libcasr/Cargo.toml:4]: Did you modify update version script?

File: libcasr/Cargo.toml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+readme = "../README.md"
+repository = "https://github.com/ispras/casr.git"
+documentation = "https://docs.rs/libcasr"
+edition = "2021"
+license = "Apache-2.0"
+exclude = ["/tests"]
+
+[dependencies]
+regex = "~1.5"
+lazy_static = "~1.4"
```
Comment (SweetVishnya) [inline libcasr/Cargo.toml:16]: dev-dependecies?

**Review by SweetVishnya — CHANGES_REQUESTED** [review]

File: casr/src/bin/casr-san.rs (hunk 3 above)
Comment (anfedotoff) [reply casr/src/bin/casr-san.rs:191]: They are defined in gdb.rs

File: libcasr/Cargo.toml (hunk 4 above)
Comment (anfedotoff) [reply libcasr/Cargo.toml:4]: yes

File: libcasr/Cargo.toml (hunk 5 above)
Comment (anfedotoff) [reply libcasr/Cargo.toml:16]: for tests

File: casr/src/bin/casr-san.rs (hunk 3 above)
Comment (SweetVishnya) [reply casr/src/bin/casr-san.rs:191]: Can we move them to libcasr

Comment (anfedotoff) [reply casr/src/bin/casr-san.rs:191]: I will move to file with constants.rs

File: update_version.sh [hunk 6]
```diff
@@ -6,10 +6,12 @@ if [ "$#" -ne 2 ]; then
...
-sed -i "s/\"$1\"/\"$2\"/g" src/bin/casr-core.rs
-sed -i "s/\"$1\"/\"$2\"/g" src/bin/casr-cli.rs
-sed -i "s/\"$1\"/\"$2\"/g" src/bin/casr-cluster.rs
-sed -i "s/\"$1\"/\"$2\"/g" src/bin/casr-san.rs
-sed -i "s/\"$1\"/\"$2\"/g" src/bin/casr-gdb.rs
-sed -i "s/\"$1\"/\"$2\"/g" src/bin/casr-afl.rs
-sed -i "s/\"$1\"/\"$2\"/g" src/bin/casr-python.rs
+sed -i "s/version = \"$1\"/version = \"$2\"/g" casr/Cargo.toml
+sed -i "s/version = \"$1\"/version = \"$2\"/g" libcasr/Cargo.toml
+sed -i "s/\"$1\"/\"$2\"/g" casr/src/bin/casr-core.rs
```
Comment (SweetVishnya) [inline update_version.sh:11]: Maybe we can use find casr-* to automatically add new tools?

**Review by SweetVishnya — CHANGES_REQUESTED** [review]

**Comment by SweetVishnya** [comment]
Add casr and libcasr crate docs. Docs for util module in casr. Update constants module docs.

File: update_version.sh (hunk 6 above)
Comment (SweetVishnya) [reply update_version.sh:11]: ok

**Review by SweetVishnya — APPROVED** [review]

**Pull request merged**